packages/engine/src/simulation/command.rs

```diff
@@ -34,32 +35,73 @@ enum HashMessageType {
     Create,
     /// Remove an Agent
     Remove,
+    /// Stop the simulation
+    Stop,
 }
 
+#[derive(Debug)]
 struct CreateCommand {
     agent: Agent,
 }
 
+#[derive(Debug)]
 struct RemoveCommand {
     uuid: Uuid,
 }
 
+/// Status of the stop message occurred.
+///
+/// See the [HASH-documentation] for more information.
+///
+/// [HASH-documentation]: https://hash.ai/docs/simulation/creating-simulations/agent-messages/built-in-message-handlers#Stopping-a-simulation
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub enum StopStatus {
+    Success,
+    Warning,
+    Error,
+}
+
+impl Default for StopStatus {
+    fn default() -> Self {
+        Self::Warning
+    }
+}
+
+/// Stop message sent from an agent.
+///
+/// See the [HASH-documentation] for more information.
+///
+/// [HASH-documentation]: https://hash.ai/docs/simulation/creating-simulations/agent-messages/built-in-message-handlers#Stopping-a-simulation
+#[derive(Debug, Default, PartialEq, Serialize, Deserialize)]
+pub struct StopMessage {
```

Comment:
Feels strange to me that we're not calling this a `StopCommand` if it's in `command.rs`. But we do call it stop_msg in other places so I don't have a strong opinion.

Reply:
I called it `StopCommand` before, but changed it to `StopMessage` as it's passed as message and also referred as message everywhere (e.g. in official docs on homepage).

Reply:
Yeah I don't like the inconsistency but I think it's the right call for now without us refactoring the other bits